app/src/main/java/org/thoughtcrime/securesms/conversation/start/newmessage/NewMessageViewModel.kt

```diff
@@ -44,7 +47,20 @@ internal class NewMessageViewModel @Inject constructor(
     }
 
     override fun onContinue() {
-        val idOrONS = state.value.newMessageIdOrOns.trim()
+        val trimmed = state.value.newMessageIdOrOns.trim()
+        // Check if all characters are ASCII (code <= 127).
+        val idOrONS = if (trimmed.all { it.code <= 127 }) {
+            // Already ASCII (or punycode‐ready); no conversion needed.
+            trimmed
+        } else {
+            try {
+                // For non-ASCII input (e.g. with emojis), attempt to puny-encode
+                IDN.toASCII(trimmed, IDN.ALLOW_UNASSIGNED)
+            } catch (e: IllegalArgumentException) {
+                // if the above failed, resort to the original trimmed string
```

Comment:
Don't fail silently.

```suggestion
                // if the above failed, resort to the original trimmed string
                Log.e("onContinue", "Punycode conversion failed for: $trimmed", e)
                trimmed
```